addons/sale_pdf_quote_builder/static/src/js/quotation_document_kanban/quotation_document_kanban_widget.js

```diff
@@ -18,9 +19,11 @@ export class QuotationDocumentX2ManyField extends X2ManyField {
     setup() {
         super.setup();
         this.uploadRoute = '/sale_pdf_quote_builder/quotation_document/upload';
-        this.formData = {
-            'sale_order_template_id': this.props.record.resId,
-        };
+        onWillRender(() => {
+            this.formData = {
+                'sale_order_template_id': this.props.record.resId,
+            };
+        });
```

Comment:
Why do you need to put it in `onWillRender`? Doesn't it work without changing anything?

Reply:
`setup` is only called when loading the page, `onWillRender` will also be called after a `web_save`, at which point you have a `resId` you can assign to the template.

Reply:
Yes, when saving a page, the records are not updated, so ``onWillRender`` was implemented to load the records.